azure-pipelines.yml

```diff
@@ -23,17 +23,53 @@ jobs:
   steps:
   - script: |
       printenv
-      echo hello
       pwd
       ls
+      echo "agend build directory"
+      echo $(Agent.BuildDirectory)
     displayName: filesystem check
   - script: |
+      cd $(Agent.BuildDirectory)/s
       git clone https://github.com/oci-labs/rosidl_typesupport_opendds.git
     displayName: clone rosidl_typesupport_opendds
   - script: |
+        cd $(Agent.BuildDirectory)
+        git clone https://github.com/adamsj-oci/rmw_build
+        vcs import s < rmw_build/ros2.repos
+        echo "ls"
+        ls
+        echo "ls s"
+        ls s/
+    displayName:  rmw build repo and dependencies
+  - script: |
+        cd $(Agent.BuildDirectory)
+        source /opt/OpenDDS/setenv.sh
+        source /opt/ros/eloquent/setup.bash
+        printenv|grep DDS
+        printenv|grep ROS
+        colcon build --symlink-install --packages-up-to rmw_opendds_cpp
+    displayName:  build up to rmw_opendds_cpp
+  - script: |
+        cd $(Agent.BuildDirectory)
+        source /opt/OpenDDS/setenv.sh
+        source /opt/ros/eloquent/setup.bash
+        source $(Agent.BuildDirectory)/install/local_setup.bash
+        printenv|grep DDS
+        printenv|grep ROS
+        colcon build --symlink-install --packages-up-to rcl_interfaces examples_rclcpp_minimal_publisher examples_rclcpp_minimal_subscriber
+    displayName:  build simple tests
+  - script: |
+        cd $(Agent.BuildDirectory)
         source /opt/OpenDDS/setenv.sh
         source /opt/ros/eloquent/setup.bash
+        source $(Agent.BuildDirectory)/install/local_setup.bash
         printenv|grep DDS
         printenv|grep ROS
-        colcon build --symlink-install
-    displayName:  environment setup and build
+        export RMW_IMPLEMENTATION=rmw_opendds_cpp
+        env|grep RMW
+        ros2 run examples_rclcpp_minimal_subscriber subscriber_member_function_twenty 1> sub.out& 
```

Comment:
This script puts a process in the background without waiting for it.  Maybe that would be a better approach than the `sleep` that's 2 lines down.

Reply:
Would it be cleaner if pub/sub exit by themselves? For example, pub sends n messages and quits. Sub receives <=n messages and quits.

Reply:
That would be good eventually, but we may not be ready for it yet.
Until then we can use external process control like wait and kill

Reply:
To be clear, these particular pub and sub applications exit automatically after 20 seconds. I have to background the subscriber so it can run concurrently with the publisher otherwise we won't see any messages received in the subscriber. The sleep 20 is required because for some reason the subscriber is taking many seconds to shutdown and make the prints available in sub.out.

Reply:
Right, so if you'd use a "wait" in the shell then we'd know that the background process is done and doesn't still have an open file handle (and data possibly buffered) for the output file.

Reply:
Much better. Thanks. I wasn't aware of the `wait` command.